### PR TITLE
Added additional logging when an Exception occurs to aid in …

### PR DESCRIPTION
**Pull request by KevinDougan**
…determining the root cause of the problem.

I found that the existing code was not displaying the error message and was swallowing the Exception without displaying the Stack Trace, so I had to debug the code to figure out what was wrong.

### CODE REVIEW

File: dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java [hunk 1]
```diff
@@ -200,7 +200,8 @@ private void stowMetaDataAndBulkData(File file) {
             try {
                 metadata = parseJSON(file.getPath());
             } catch (Exception e) {
-                LOG.error("error parsing metadata JSON file {}", file);
+                LOG.error("error parsing metadata JSON file {}: {}", file, e.getMessage());
+                e.printStackTrace();
```
Comment (hczedik) [inline dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java]: Hi, Kevin!
As we are already using a Logger here, I wouldn't use e.printStackTrace() (which will print to standard out).
I believe if you do it like this the stack trace should also be visible:
`LOG.error("error parsing metadata JSON file {}: {}", file, e);`
Can you try that out please.

**Comment by hczedik** [comment]
Please do not merge yet, see my inline-comment discussion first.

**Comment by KevinDougan** [comment]
Your suggestion has been added.

File: dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ private void stowMetaDataAndBulkData(File file) {
             try {
                 metadata = parseJSON(file.getPath());
             } catch (Exception e) {
-                LOG.error("error parsing metadata JSON file {}", file);
+                LOG.error("error parsing metadata JSON file {}: {}", file, e);
```
Comment (hczedik) [inline dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java]: For slf4j loggers will display the full exception, including stacktrace, even if you do not add the placeholder to the message itself:
`LOG.error("error parsing metadata JSON file {}", file, e);`
Please give that a try.

**Comment by KevinDougan** [comment]
Hi there,

When I use your latest suggested format, it does not print the Stack Trace, but only prints the message. I think it's more valuable to have the Stack Trace there, especially since this is an Error situation, so I think it would be good to go with the currently committed format. Let me know if you agree.

Thanks,
Kevin

**Comment by KevinDougan** [comment]
OK, I just pushed another change up, so it should be good to go now.

Thanks,
Kevin

**Pull request merged**